client/network/src/service.rs

```diff
@@ -262,24 +268,106 @@ where
 			)
 		};
 
-		let (protocol, peerset_handle, mut known_addresses) = Protocol::new(
+		let peer_store = PeerStore::new(
+			network_config.boot_nodes.iter().map(|bootnode| bootnode.peer_id).collect(),
+		);
+		let peer_store_handle = peer_store.handle();
+
+		// Spawn `PeerStore` runner.
+		(params.executor)(peer_store.run().boxed());
```

Comment:
I'm not happy with using the executor originally intended for `Swarm`, but pulling a `sc-service` crate dependency for `SpawnTaskHandle` seemed even worse.